src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,35 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new IllegalArgumentException("User cannot be null");
+        }
+
+        // Check if user already exists in the database
+        for (User existingUser : storageDao.getAll()) {
+            if (existingUser.getLogin().equals(user.getLogin())) {
+                throw new IllegalArgumentException("User with this login already exists");
+            }
+        }
+
+        // Validate user parameters
+        if (user.getLogin() == null || user.getLogin().isEmpty() || user.getLogin().length() < 6) {
```

Comment:
![image](https://github.com/mate-academy/jv-registration-validation/assets/20326858/edb7f5db-f337-48c1-a043-e1a0be68e010)

```suggestion
        if (user.getLogin() == null || user.getLogin().length() < MIN_LOGIN_LENGTH) {
```